src/smoothing/SmoothingAndStep.java

```diff
@@ -80,6 +80,10 @@ public double smoothing(int[] params, String inPath, String outPath,
 
                 if((strLine = br.readLine()) != null){
                     while(!addLine(strLine)){
+                        if(Thread.currentThread().isInterrupted()){
```

Comment:
I'll look into a better solution once I have the time (such as running it as an external process).